Sources/PostgresNIO/New/PSQLConnection.swift

```diff
@@ -215,17 +215,16 @@ final class PSQLConnection {
                     .channelInitializer { channel in
                         let decoder = ByteToMessageHandler(PSQLBackendMessage.Decoder())
                         
-                        var enableSSLCallback: ((Channel) -> EventLoopFuture<Void>)? = nil
+                        var enableSSLCallback: ((Channel) throws -> ())? = nil
                         if let tlsConfiguration = configuration.tlsConfiguration {
                             enableSSLCallback = { channel in
-                                channel.eventLoop.submit {
-                                    let sslContext = try NIOSSLContext(configuration: tlsConfiguration)
-                                    return try NIOSSLClientHandler(
-                                        context: sslContext,
-                                        serverHostname: configuration.sslServerHostname)
-                                }.flatMap { sslHandler in
-                                    channel.pipeline.addHandler(sslHandler, position: .before(decoder))
-                                }
+                                channel.eventLoop.preconditionInEventLoop()
```

Comment:
FWIW the sync operations will do this check for you; keeping this (and potentially degrading it to an assertion) is a useful as an indicator to whoever looks at this code though.